networkx/algorithms/traversal/depth_first_search.py

```diff
@@ -53,38 +51,42 @@ def dfs_edges(G, source=None):
     else:
         # produce edges for components with source
         nodes = [source]
-    visited=set()
+    visited = set()
+    if depth_limit is None:
+       depth_limit = len(G)
     for start in nodes:
         if start in visited:
             continue
         visited.add(start)
-        stack = [(start,iter(G[start]))]
+        stack = [(start, depth_limit, iter(G[start]))]
         while stack:
-            parent,children = stack[-1]
+            parent, depth_now, children = stack[-1]
+            if not depth_now >= 1:
```

Comment:
```if depth_now < 1:``` or maybe ```if depth_now == 0:```

Also, would it be more efficient to check depth before putting the child in the stack?